### PR TITLE
Add sass tests too

### PR DESCRIPTION
**Pull request by XhmikosR**
@xzyfer I think it's worth having this. That being said, I noticed a few differences while working on this patch. This one makes me wonder:

https://github.com/dlmanning/gulp-sass/pull/842/files#diff-5a684c173d69bcc8d9e0e4b50ac83d8eaf640dbd8e141c14d9e1a953c87afbd3R482

### CODE REVIEW

**Comment by XhmikosR** [comment]
@xzyfer I think it's worth having this. That being said, I noticed a few differences while working on this patch. This one makes me wonder:

https://github.com/dlmanning/gulp-sass/pull/842/files#diff-5a684c173d69bcc8d9e0e4b50ac83d8eaf640dbd8e141c14d9e1a953c87afbd3R482

**Comment by XhmikosR** [comment]
Oh, and if you have another idea how to implement this; I went with the environment variable solution.

File: test/scss/includes/_dogs.sass [hunk 1]
```diff
@@ -1,5 +1,5 @@
-$blue: #3bbfce;
-$margin: 16px;
+$blue: #3bbfce
```
Comment (XhmikosR) [inline test/scss/includes/_dogs.sass:1]: dart sass threw an error for indented syntax with semicolons

File: test/main.js [hunk 2]
```diff
@@ -242,25 +254,25 @@ describe('gulp-sass -- async compile', () => {
       'includes/_dogs.sass',
     ];
 
-    const stream = sass();
+    const stream = gulpSass();
     stream.on('data', (cssFile) => {
       assert.ok(cssFile.sourceMap);
-      assert.deepEqual(cssFile.sourceMap.sources, expectedSources);
+      assert.deepEqual(cssFile.sourceMap.sources.sort(), expectedSources.sort());
```
Comment (XhmikosR) [inline test/main.js:260]: I had to use `sort()` since dart sass and node-sass produce different sorting.

File: test/main.js [hunk 3]
```diff
@@ -12,7 +12,12 @@ const autoprefixer = require('autoprefixer');
 const tap = require('gulp-tap');
 const globule = require('globule');
 
-const sass = require('../index')(require('node-sass'));
+const COMPILER = process.env.SASS_COMPILER === 'sass' ? 'sass' : 'node-sass';
+
+// eslint-disable-next-line import/no-dynamic-require
+const gulpSass = require('../index')(require(COMPILER));
```
Comment (xzyfer) [inline test/main.js:18]: FWIW reducing unnecessary diff noise lowers the burden on reviewers.

```suggestion
const sass= require('../index')(require(COMPILER));
```

Comment (XhmikosR) [reply test/main.js:18]: I'm aware but initially I was planning to use a `sass` variable for dart sass. I guess now I can revert this change.

File: test/main.js [hunk 4]
```diff
@@ -458,19 +479,21 @@ describe('gulp-sass -- sync compile', () => {
       'inheritance.scss',
     ];
 
+    if (COMPILER === 'sass') expectedSourcesAfter.push('inheritance.css');
```
Comment (XhmikosR) [inline test/main.js:482]: This makes me wonder if something else is wrong... Dart sass includes the dist file while node-sass doesn't.

File: test/main.js [hunk 5]
```diff
@@ -157,13 +162,17 @@ describe('gulp-sass -- async compile', () => {
     const stream = sass();
 
     stream.on('error', (err) => {
-      // Error must include message body
```
Comment (xzyfer) [inline test/main.js:160]: We use this style because error message formats are not part of the sass language spec and change over time. This style is more resilient to compiler changes as it simple asserts for key phrases.

File: test/main.js (hunk 4 above)
Comment (xzyfer) [reply test/main.js:482]: Source maps are not part of the sass language spec and there are difference between engines. It's likely node-sass is wrong in this case. A more resilient approach might be to check for the intersection of the sources rather than a matching setting.

File: test/main.js (hunk 5 above)
Comment (XhmikosR) [reply test/main.js:160]: Yeah, but it's more troublesome to keep them in sync. For example, how should I test for the line with dart sass? Just check for number `2`?

Comment (xzyfer) [reply test/main.js:160]: I'd opt for simple removing the line number. This is more the concern of the engine. All we really care about is that an error was produced.

Comment (XhmikosR) [reply test/main.js:160]: Please check now, it should be what you want. The line is still checked in another test, not sure if we should remove that too.

File: test/main.js (hunk 4 above)
Comment (XhmikosR) [reply test/main.js:474]: Feel free to push your suggestion if you want otherwise let's land it as is.

File: package.json [hunk 6]
```diff
@@ -10,7 +10,9 @@
     "lint": "eslint --report-unused-disable-directives --ignore-path .gitignore .",
     "fix": "npm run lint -- --fix",
     "mocha": "mocha",
-    "test": "mocha"
+    "test": "npm run test:node-sass && npm run test:dart-sass",
+    "test:node-sass": "mocha",
```
Comment (XhmikosR) [inline package.json:14]: Let me know if you want different script names here.

File: package.json [hunk 7]
```diff
@@ -10,7 +10,9 @@
...
+    "test:dart-sass": "cross-env SASS_COMPILER=sass mocha"
```
Comment (xzyfer) [inline package.json:15]: Out of curiosity could you do something like `mocha -- sass` and pull the compiler off of `argv` instead of introducing cross-env?

Comment (XhmikosR) [reply package.json:15]: It could probably work (not `mocha -- sass` but `mocha -- --sass`, but seems more hacky from a quick look. Feel free to tweak it, though.

Comment (XhmikosR) [reply package.json:15]: Actually, wait. Let me try something.

Comment (XhmikosR) [reply package.json]: I pushed a patch which does work but still seems hacky, but correct me if I'm wrong. Aren't we abusing the fact that mocha doesn't throw on unknown command line options?

Comment (xzyfer) [reply package.json]: I'm fine with it until it's a problem :)

**Comment by XhmikosR** [comment]
@xzyfer how about releasing a new minor version? We've landed plenty of changes and personally I don't have any further ones. :)

**Comment by xzyfer** [comment]
v5.1.0 has been released
